gpkitmodels/GP/aircraft/prop/propeller.py

```diff
@@ -20,37 +24,61 @@ class Propeller_Performance(Model):
     CP                      [-]         power coefficient
     Q                       [N*m]       torque
     omega                   [rpm]       propeller rotation rate
-
+    omega_max  10000        [rpm]       max rotation rate
+    P_shaft                 [kW]        shaft power
+    M_tip      .5           [-]         Tip mach number
+    a          295          [m/s]       Speed of sound at altitude
     """
 
     def helper(self, c):
-        " helper function to avoid signomial constraint "
         return 2. - 1./c[self.etaadd]
 
-    def setup(self, static, state):
-        exec parse_variables(Propeller_Performance.__doc__)
+    def setup(self, static, state = None, rho = None, V = None):
+        exec parse_variables(ActuatorProp.__doc__)
 
         V = state.V
         rho = state.rho
         R = static.R
 
-        return [eta <= etav*etai,
-                Tc == T/(0.5*rho*V**2*pi*R**2),
-                z2 >= Tc + 1,
-                etai*(z1 + z2**0.5/etaadd) <= 2,
-               ], static
+        constraints = [eta <= etav*etai,
+                       Tc >= T/(0.5*rho*V**2*pi*R**2),
+                       z2 >= Tc + 1,
+                       etai*(z1 + z2**0.5/etaadd) <= 2,
+                       lam >= V/(omega*R),
+                       CT >= Tc*lam**2,
+                       CP <= Q*omega/(.5*rho*(omega*R)**3*pi*R**2),
+                       eta >= CT*lam/CP,
```

Comment:
Does this over constrain `eta`?

Reply:
No; eta is determined by eta_v and eta_i and then used to constrain the ratio of output to input power (CT lam)/(CP)

Reply:
Oh right.  It was over constrained in the other one.